internal/resources/providers/azurelib/inventory/storage_provider.go

```diff
@@ -152,6 +180,62 @@ func (p *storageAccountProvider) ListStorageAccountBlobServices(ctx context.Cont
 	return assets, nil
 }
 
+func (p *storageAccountProvider) ListStorageAccountQueues(ctx context.Context, storageAccounts []AzureAsset) ([]AzureAsset, error) {
+	var assets []AzureAsset
+	for _, sa := range storageAccounts {
+		responses, err := p.client.AssetQueues(ctx, sa.SubscriptionId, nil, sa.ResourceGroup, sa.Name, nil)
+		if err != nil {
+			return nil, fmt.Errorf("error while fetching azure queues for storage accounts %s: %w", sa.Id, err)
+		}
+
+		queues, err := transformQueues(responses, sa)
+		if err != nil {
+			return nil, fmt.Errorf("error while transforming azure queues for storage accounts %s: %w", sa.Id, err)
+		}
+
+		assets = append(assets, queues...)
+	}
+
+	return assets, nil
+}
+
+func (p *storageAccountProvider) ListStorageAccountQueueServices(ctx context.Context, storageAccounts []AzureAsset) ([]AzureAsset, error) {
+	var assets []AzureAsset
+	for _, sa := range storageAccounts {
+		response, err := p.client.AssetQueueServices(ctx, sa.SubscriptionId, nil, sa.ResourceGroup, sa.Name, nil)
+		if err != nil {
+			return nil, fmt.Errorf("error while fetching azure queue services for storage accounts %s: %w", sa.Id, err)
+		}
+
+		var errs error
+		for _, item := range response.Value {
+			properties, err := maps.AsMapStringAny(item.QueueServiceProperties)
+			if err != nil {
+				errs = errors.Join(errs, err)
+			}
+
+			assets = append(assets, AzureAsset{
+				Id:             pointers.Deref(item.ID),
+				Name:           pointers.Deref(item.Name),
+				Type:           strings.ToLower(pointers.Deref(item.Type)),
+				ResourceGroup:  sa.ResourceGroup,
+				SubscriptionId: sa.SubscriptionId,
+				TenantId:       sa.TenantId,
+				Properties:     properties,
+				Extension: map[string]any{
+					ExtensionStorageAccountID:   sa.Id,
+					ExtensionStorageAccountName: sa.Name,
+				},
+			})
+		}
+		if errs != nil {
+			return nil, fmt.Errorf("error while transforming azure queue services for storage accounts %s: %w", sa.Id, errs)
```

Comment:
I wonder if we should return nil here and don't store any StorageQueueService. 

The error is not a blocking one, right? We rather have the asset without properties anyway - and log the error.

Wdyt?

Reply:
Makes sense 👍